python/fapolicy_analyzer/ui/__main__.py

```diff
@@ -6,23 +6,50 @@
 gi.require_version("Gtk", "3.0")
 from gi.repository import Gtk
 from .main_window import MainWindow
+from .state_manager import stateManager
+
+# Globals
+gstrEditSessionFileName = None
+gbAutosaveEnabled = False
 
 
 def parse_cmdline():
+    global gstrEditSessionFileName
+    global gbAutosaveEnabled
+
     parser = argparse.ArgumentParser()
     parser.add_argument("-v", "--verbose", action="store_true",
                         help="Enable verbose mode")
+    parser.add_argument("-a", "--autosave", action="store_true",
+                        help="Enable edit session autosave mode")
+    parser.add_argument("-s", "--session",
+                        help="Specify edit session tmp file basename")
     args = parser.parse_args()
 
     # Set Verbosity Level
     if args.verbose:
         logging.root.setLevel(logging.DEBUG)
         logging.debug("Verbosity enabled.")
 
+    # Enable edit session autosaves
+    if args.autosave:
+        gbAutosaveEnabled = True
+
+    # Set Edit Session Tmp File
+    if args.session:
+        gstrEditSessionFileName = args.session
+
+    # Return the parsed args to simplify unit-testing
+    dictArgs = dict(vars(args))
+    return dictArgs
+
 
 def main():
     parse_cmdline()
     MainWindow()
+    stateManager.set_autosave_enable(gbAutosaveEnabled)
+    if gstrEditSessionFileName:
+        stateManager.set_autosave_filename(gstrEditSessionFileName)
     Gtk.main()
```

Comment:
or return a dict like 
```
return {
  "autosave": args.autosave,
  "session": args.session
}
```
and use that in the `main()` function.

Reply:
I'm partially to having functions only do more or less what their name state, i.e. no side effects; I think it's easier to maintain, reverse-engineer, extend. Granted it's in the same file and in the function immediately above it's use in this particular case. 

I'm ok with getting rid of the globals for a returned dict. I'll make those changes and we'll give it another review spin. Thx.

Reply:
I went with your general suggestions, i.e. calling the stateManager accessors from within parse_cmdline(). Otherwise main() gets ugly...There's still a few things I'm going to modify before a final commit.